### ISSUE TITLE
Type-cast of variables not written to process.env

### ISSUE THREAD

**Issue by andyburton** [niftylettuce/dotenv-parse-variables#8]
The README gives an example with `console.log(process.env);` which outputs type-casted variables e.g. BAZ: 2 vs BAZ: '2'.

This is not the case - any variable written back to process.env will be output as a string, and it does not appear possible to set and then read a type-casted variable from process.env.

The assignToProcessEnv and overrideProcessEnv work, they set the value to process.env as per the logic, but the type is lost.

You need to use the output of dotenvExtended.load() to see the type-casted variables, however this does not contain anything existing from process.env - so there is no place to get an entire process.env variable list with type-cast.

The way i have achieved this is by passing a copy of process.env into the package e.g.

```const dotenvExtended = require('dotenv-extended');
const loadedDotEnv = dotenvExtended.load({
  silent: false,
  errorOnMissing: true
});
const env = require('dotenv-parse-variables')(Object.assign({}, process.env));

console.log(env);

**Comment by jcblw** [comment]
Hey @andyburton 

Contributor to [dotenv](https://github.com/motdotla/dotenv), and was responding to an issue [#342](https://github.com/motdotla/dotenv/issues/342) on the dotenv repo. I noticed that it is mentioned in the documentation of this plugin that this does work. I looked at the tests and it seems like there are no tests for this type of scenario. From working with the `process.env` in the past I have noticed that everything gets casted to string when added to the `process.env`.

For example
```shell
❯ node -e "process.env.FOO = ['foo', 'bar']; console.log(typeof process.env.FOO);"
string
```

So to be able to keep your types you will need to pass along a separate object with these values.
